openpype/hosts/blender/plugins/create/create_blendScene.py

```diff
@@ -2,13 +2,11 @@
 
 import bpy
 
-from openpype.pipeline import get_current_task_name
-from openpype.hosts.blender.api import plugin, lib, ops
-from openpype.hosts.blender.api.pipeline import AVALON_INSTANCES
+from openpype.hosts.blender.api import plugin, lib
 
 
 class CreateBlendScene(plugin.Creator):
```

Comment:
```suggestion
class CreateBlendScene(plugin.BaseCreator):
    """Generic group of assets."""

    identifier = "io.openpype.creators.blender.blendScene"
    name = "blendScene"
    label = "Blender Scene"
    family = "blendScene"
    icon = "cubes"

    maintain_selection = False

    def _create(self, instance_node):
        selection = lib.get_selection(include_collections=True)
        for data in selection:
            if isinstance(data, bpy.types.Collection):
                instance_node.children.link(data)
            elif isinstance(data, bpy.types.Object):
                instance_node.objects.link(data)


    def create(
        self, subset_name: str, instance_data: dict, pre_create_data: dict
    ):

        instance_node = super().create(subset_name,
                                       instance_data,
                                       pre_create_data)

        if pre_create_data.get("use_selection"):
            mti = ops.MainThreadItem(self._create, instance_node)
            ops.execute_in_main_thread(mti)

        return instance_node
```

This creator still needs to run in the main thread, because to select collections we need to access the outliner, and it isn't enough to change context, because from the Qt window of the creator we cannot access the main Blender window, AFAIK!

I also changed the creator to inherit from the new Blender Creator and added the identifier.

Reply:
The creator worked fine for me. Could you pull the branch and give it a go?

Reply:
I've tried from the latest commit 😅 if I don't change the inherited class, I cannot even see the family in the creator!

Reply:
You're right about the parent class, it's my bad for missing it. But I don't see why you would need to execute in main thread. What version of Blender are you testing on?

Reply:
3.6! I get this error without changes:
```
Creator: Blender Scene (<property object at 0x000001A82D6BAD40>)

Error: 'NoneType' object has no attribute 'screen'

Traceback (most recent call last):
  File "C:\Dev\Ynput\OpenPype\openpype\pipeline\create\context.py", line 2036, in _create_with_unified_error
    result = creator.create(*args, **kwargs)
  File "C:\Dev\Ynput\OpenPype\openpype\hosts\blender\plugins\create\create_blendScene.py", line 27, in create
    selection = lib.get_selection(include_collections=True)
  File "C:\Dev\Ynput\OpenPype\openpype\hosts\blender\api\lib.py", line 317, in get_selection
    selection.extend(get_selected_collections())
  File "C:\Dev\Ynput\OpenPype\openpype\hosts\blender\api\lib.py", line 283, in get_selected_collections
    area for area in bpy.context.window.screen.areas
AttributeError: 'NoneType' object has no attribute 'screen'
```

Reply:
What about using `bpy.context.screen` instead?

Reply:
@simonebarbieri could you try again?

Reply:
@BigRoy oh let me try again!

Reply:
Oh yes, it works :) @BigRoy That's great! I'll try to break it!

Reply:
Ah @BigRoy the creator still needs an identifier!

Reply:
Fixed with https://github.com/ynput/OpenPype/pull/5910/commits/f281da5784f79cacda9730c5b211a77e48bfeb2b ? :)

Reply:
Also I believe we can remove all the `name` attributes from the new style creators as well, right?

Reply:
It doesn't work if the outliner is not in the main window, but the error that it throws already warns about that. And this was the same even before the new publisher anyway.
I think it is fine to keep it like this for now, because technically we could have even different selections in different outliners...
<img alt="image" src="https://github.com/ynput/OpenPype/assets/1087869/ff8ace55-9d6f-4a24-93fb-cf84a5412253">
While the selection of objects is actually on the object, and so independent from the outliner. So for now I would keep it like this, and hope that Blender implements a proper way to select and get selected collections.

Reply:
> Also I believe we can remove all the `name` attributes from the new style creators as well, right?

Yes I don't think there's any use for it with the new publisher.

Reply:
Ok, will mark this thread resolved for now - feel free to unresolve if other issues appear.